api/app/signals/apps/signals/admin.py

```diff
@@ -215,6 +216,19 @@ class RoutingExpressionAdmin(admin.ModelAdmin):
     list_filter = ['_expression___type__name', '_department__code']
     list_display = ['_expression', '_department', 'is_active']
 
+    def save_model(self, request, obj, form, change):
+        # if is_active is true, try to compile code, deactivate when invalid
+        try:
+            if obj.is_active:
+                expr = form.cleaned_data.get('_expression', None)
+                if expr:
+                    ExpressionEvaluator().compile(expr.code)
+        except Exception as e:
+            obj.is_active = False
+            messages.add_message(request, messages.WARNING, f'Rule deactivated due to error in expression: {str(e)}')
+
+        super(RoutingExpressionAdmin, self).save_model(request, obj, form, change)
+
```

Comment:
Should `_expression` being `None` also lead to an inactive routing expression?

Reply:
We can't store empty code expression, it's required field, so this should be ok.